### DISCUSSION TITLE
Boot environment deletion

### DISCUSSION THREAD

**Discussion by Sithuk** [zbm-dev/zfsbootmenu#354] [General]
What is the appropriate method to delete boot environments? I recall that one of the team mentioned that zfsbootmenu is a read only environment and that allowing BE deletion would go against that philosophy. However, what about at the command line once booted into the system? I have used zectl in the past, but wondered if there was a tool I didn't know about. Perhaps from upstream openzfs?

**Comment by ahesford** [comment]
There isn't really a standard definition of "boot environment". To ZBM, a boot environment is just a root filesystem that has a kernel it can launch. From the running system, just `zfs destroy` the filesystem you don't want. You can also do this from ZBM. In the menu, select the boot environment you want to destroy. Press `CTRL+W` to reimport the pool read/write, then `CTRL+R` to drop to a shell. You can then `zfs destroy` whatever you like.
